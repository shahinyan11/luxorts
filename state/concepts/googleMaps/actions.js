import * as types from './types';

export const fetchGeocoder = ({ query, reverse = false }) => ({
  type: types.FETCH_GEOCODER,
  query,
  reverse,
});

export const setGeocoder = (items) => ({
  type: types.SET_GEOCODER,
  items,
});

export const clearGeocoder = () => ({
  type: types.CLEAR_GEOCODER,
});
