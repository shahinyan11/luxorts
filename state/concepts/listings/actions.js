import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';

import * as types from './types';

export const createUpdateListing = makeFormSubmitAction(types.CREATE_UPDATE_LISTING);

export const fetchListings = ({ skipLoading = false, skipClearing = false } = {}) => ({
  type: types.FETCH_LISTINGS,
  skipLoading,
  skipClearing,
});

export const setListingsPage = (pageNumber) => ({
  type: types.SET_LISTINGS_PAGE,
  pageNumber,
});

export const setListingsSortOrder = (sortKey) => ({
  type: types.SET_LISTINGS_SORT_ORDER,
  sortKey,
});

export const setListingTableSettings = (settings) => ({
  type: types.SET_LISTING_TABLE_SETTINGS,
  settings,
});

export const filterListings = (filters) => ({
  type: types.FILTER_LISTINGS,
  filters,
});

export const setListingsFilterParams = (filterParams) => ({
  type: types.SET_LISTINGS_FILTER_PARAMS,
  filterParams,
});

export const setListingsSearchQuery = (query) => ({
  type: types.SET_LISTINGS_SEARCH_QUERY,
  query,
});

export const fetchListing = (listingId) => ({
  type: types.FETCH_LISTING,
  listingId,
});

export const fetchListingPropertyType = (listingId) => ({
  type: types.FETCH_LISTING_PROPERTY_TYPE,
  listingId,
});

export const fetchListingsBasePropertyTypes = () => ({
  type: types.FETCH_LISTINGS_BASE_PROPERTY_TYPES,
});

export const fetchListingAccomodation = (listingId) => ({
  type: types.FETCH_LISTING_ACCOMODATION,
  listingId,
});

export const updateListingAccomodation = makeFormSubmitAction(types.UPDATE_LISTING_ACCOMODATION);

export const fetchListingLocation = (listingId) => ({
  type: types.FETCH_LISTING_LOCATION,
  listingId,
});

export const updateListingLocation = makeFormSubmitAction(types.UPDATE_LISTING_LOCATION);

export const fetchListingBaseAmenities = () => ({
  type: types.FETCH_LISTING_BASE_AMENITIES,
});

export const fetchListingAmenities = (listingId) => ({
  type: types.FETCH_LISTING_AMENITIES,
  listingId,
});

export const updateListingAmenities = makeFormSubmitAction(types.UPDATE_LISTING_AMENITIES);

export const fetchListingBaseCapabilities = () => ({
  type: types.FETCH_LISTING_BASE_CAPABILITIES,
});

export const fetchListingCapabilities = (listingId) => ({
  type: types.FETCH_LISTING_CAPABILITIES,
  listingId,
});

export const updateListingCapabilities = makeFormSubmitAction(types.UPDATE_LISTING_CAPABILITIES);

export const fetchListingBaseServices = () => ({
  type: types.FETCH_LISTING_BASE_SERVICES,
});

export const fetchListingServices = (listingId) => ({
  type: types.FETCH_LISTING_SERVICES,
  listingId,
});

export const updateListingServices = makeFormSubmitAction(types.UPDATE_LISTING_SERVICES);

export const fetchListingPhotos = (listingId) => ({
  type: types.FETCH_LISTING_PHOTOS,
  listingId,
});

export const updateListingPhotos = makeFormSubmitAction(types.UPDATE_LISTING_PHOTOS);

export const fetchListingDescription = (listingId) => ({
  type: types.FETCH_LISTING_DESCRIPTION,
  listingId,
});

export const updateListingDescription = makeFormSubmitAction(types.UPDATE_LISTING_DESCRIPTION);

export const fetchListingBaseHouseRules = () => ({
  type: types.FETCH_LISTING_BASE_HOUSE_RULES,
});

export const fetchListingHouseRules = (listingId) => ({
  type: types.FETCH_LISTING_HOUSE_RULES,
  listingId,
});

export const updateListingHouseRules = makeFormSubmitAction(types.UPDATE_LISTING_HOUSE_RULES);

export const fetchListingReservationPreferences = (listingId) => ({
  type: types.FETCH_LISTING_RESERVATION_PREFERENCES,
  listingId,
});

export const updateListingReservationPreferences = makeFormSubmitAction(
  types.UPDATE_LISTING_RESERVATION_PREFERENCES,
);

export const fetchListingTimePeriods = ({ listingId, startDate, endDate }) => ({
  type: types.FETCH_LISTING_TIME_PERIODS,
  listingId,
  startDate,
  endDate,
});

export const createListingTimePeriods = ({
  listingId,
  startDate,
  endDate,
  fetchStartDate,
  fetchEndDate,
  kind,
}) => ({
  type: types.CREATE_LISTING_TIME_PERIODS,
  listingId,
  startDate,
  endDate,
  fetchStartDate,
  fetchEndDate,
  kind,
});

export const fetchListingPricing = (listingId) => ({
  type: types.FETCH_LISTING_PRICING,
  listingId,
});

export const updateListingPricing = makeFormSubmitAction(types.UPDATE_LISTING_PRICING);

export const updateListingStatus = ({ listingId, status, modal, message, hideModal }) => ({
  type: types.UPDATE_LISTING_STATUS,
  listingId,
  status,
  modal,
  message,
  hideModal,
});

export const updateListingAvailability = ({ listingId, redirectRoute }) => ({
  type: types.UPDATE_LISTING_AVAILABILITY,
  listingId,
  redirectRoute,
});
