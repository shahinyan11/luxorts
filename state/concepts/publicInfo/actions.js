import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';

import * as types from 'state/concepts/publicInfo/types';

export const fetchPopularApartments = () => ({
  type: types.FETCH_POPULAR_APARTMENTS,
});

export const fetchPropertyTypes = () => ({
  type: types.FETCH_PROPERTY_TYPES,
});

export const fetchTrendingLocations = () => ({
  type: types.FETCH_TRENDING_LOCATIONS,
});

export const fetchHomePageData = () => ({
  type: types.FETCH_HOME_PAGE_DATA,
});

export const setApartmentsPage = (pageNumber) => ({
  type: types.SET_APARTMENTS_PAGE,
  pageNumber,
});

export const fetchAboutUsPageData = () => ({
  type: types.FETCH_ABOUT_US_PAGE_DATA,
});

export const fetchTermsPageData = () => ({
  type: types.FETCH_TERMS_PAGE_DATA,
});

export const contactUsRequest = makeFormSubmitAction(types.CONTACT_US_REQUEST);

export const fetchPrivacyPolicyPageData = () => ({
  type: types.FETCH_PRIVACY_POLICY_PAGE_DATA,
});
