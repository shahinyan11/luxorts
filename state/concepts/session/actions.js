import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';
import * as types from 'state/concepts/session/types';

export const userSignUp = makeFormSubmitAction(types.USER_SIGN_UP);

export const userSignInSuccess = (currentUser) => ({
  type: types.USER_SIGN_IN_SUCCESS,
  currentUser,
});

export const userSignOut = () => ({
  type: types.USER_SIGN_OUT,
});

export const userSignOutSuccess = () => ({
  type: types.USER_SIGN_OUT_SUCCESS,
});

export const verifyUserEmail = makeFormSubmitAction(types.VERIFY_USER_EMAIL);

export const verifyUserPhone = makeFormSubmitAction(types.VERIFY_USER_PHONE);

export const resendUserEmailVerification = () => ({
  type: types.RESEND_USER_EMAIL_VERIFICATION,
});

export const userSignIn = makeFormSubmitAction(types.USER_SIGN_IN);

export const userActivateAccount = ({ redirectRoute = null, isModal = false }) => ({
  type: types.USER_ACTIVATE_ACCOUNT,
  redirectRoute,
  isModal,
});

export const updateUser = (userData) => ({
  type: types.UPDATE_USER,
  userData,
});

export const userSendForgotPasswordEmail = makeFormSubmitAction(
  types.USER_SEND_FORGOT_PASSWORD_EMAIL,
);

export const userValidateForgotPasswordCode = makeFormSubmitAction(
  types.USER_VALIDATE_FORGOT_PASSWORD_CODE,
);

export const userResendForgotPasswordEmail = () => ({
  type: types.USER_RESEND_FORGOT_PASSWORD_EMAIL,
});

export const userUpdatePassword = makeFormSubmitAction(types.USER_UPDATE_PASSWORD);

export const userUpdateAvatar = (file) => ({
  type: types.USER_UPDATE_AVATAR,
  file,
});

export const userRemoveAvatar = () => ({
  type: types.USER_REMOVE_AVATAR,
});

export const userUpdateProfile = makeFormSubmitAction(types.USER_UPDATE_PROFILE);

export const userDeactivateAccount = makeFormSubmitAction(types.USER_DEACTIVATE_ACCOUNT);

export const userChangePassword = makeFormSubmitAction(types.USER_CHANGE_PASSWORD);

export const userInviteFriends = makeFormSubmitAction(types.USER_INVITE_FRIENDS);

export const fetchUserInvitation = () => ({
  type: types.FETCH_USER_INVITATION,
});

export const setInvitationsPage = (pageNumber) => ({
  type: types.SET_INVITATIONS_PAGE,
  pageNumber,
});

export const userResendInvite = (invitationId) => ({
  type: types.USER_RESEND_INVITE,
  invitationId,
});

export const userSendUpdatePhone = makeFormSubmitAction(types.USER_SEND_UPDATE_PHONE);

export const userResendUpdatePhone = () => ({
  type: types.USER_RESEND_UPDATE_PHONE,
});

export const userUpdatePhone = makeFormSubmitAction(types.USER_UPDATE_PHONE);

export const userResendUpdateEmail = () => ({
  type: types.USER_RESEND_UPDATE_EMAIL,
});

export const userSendUpdateEmail = makeFormSubmitAction(types.USER_SEND_UPDATE_EMAIL);

export const userUpdateEmail = makeFormSubmitAction(types.USER_UPDATE_EMAIL);
