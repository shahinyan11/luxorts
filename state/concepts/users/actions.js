import * as types from 'state/concepts/users/types';

// eslint-disable-next-line import/prefer-default-export
export const fetchSelf = () => ({
  type: types.FETCH_SELF,
});
