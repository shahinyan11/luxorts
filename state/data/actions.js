import * as types from 'state/data/types';

export const dataApiRequest = ({ endpoint }) => ({
  type: types.DATA_API_REQUEST,
  endpoint,
});

export const dataApiSuccess = ({ endpoint, response }) => ({
  type: types.DATA_API_SUCCESS,
  endpoint,
  response,
});

export const dataApiFailure = ({ endpoint }) => ({
  type: types.DATA_API_FAILURE,
  endpoint,
});

export const dataDelete = ({ kind, ids }) => ({
  type: types.DATA_DELETE,
  kind,
  ids,
});

export const dataRemoveRelationship = ({ kind, ownerId, relationship, ids }) => ({
  type: types.DATA_REMOVE_RELATIONSHIP,
  kind,
  ownerId,
  relationship,
  ids,
});

export const dataDeleteEntity = (kind) => ({
  type: types.DATA_DELETE_ENTITY,
  kind,
});
