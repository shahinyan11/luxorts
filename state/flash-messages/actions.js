import { v4 as uuid } from 'uuid';

import * as types from './types';
import { MESSAGE_TYPE, MESSAGE_DURATION } from './messagesTypes';

export const showMessage = ({
  id = uuid(),
  messageType = MESSAGE_TYPE.SUCCESS,
  duration = MESSAGE_DURATION.DEFAULT,
  description,
}) => ({
  type: types.SHOW_MESSAGE,
  id,
  messageType,
  description,
  duration,
});

export const hideMessage = (id) => ({
  type: types.HIDE_MESSAGE,
  id,
});

export const hideAll = () => ({
  type: types.HIDE_ALL,
});
