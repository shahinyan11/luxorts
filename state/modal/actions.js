import * as types from 'state/modal/types';

export const showModal = ({ modalType, modalProps }) => ({
  type: types.SHOW_MODAL,
  modalType,
  modalProps,
});

export const hideModal = () => ({
  type: types.HIDE_MODAL,
});
