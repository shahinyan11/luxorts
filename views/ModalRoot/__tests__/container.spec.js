import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { hideModal } from 'state/modal/actions';
import diveTo from 'utils/testHelpers/diveToEnzyme';
import ModalRootWrapped, { ModalRootContainer } from '../container';

jest.mock('state/modal/actions', () => ({
  hideModal: jest.fn(),
}));

describe('ModalRoot container', () => {
  const store = configureStore()({
    modal: {
      modalType: 'MY_MODAL',
      modalProps: {
        customProp: 'customProp',
      },
    },
  });
  store.dispatch = jest.fn();

  const wrapper = shallow(<ModalRootWrapped store={store} history={{}} />);
  const container = diveTo(wrapper, ModalRootContainer);

  it('renders ModalRoot component', () => {
    expect(container).toMatchSnapshot();
  });

  it('onClose()', () => {
    container.props().onClose();
    expect(hideModal).toHaveBeenCalledTimes(1);
  });
});
