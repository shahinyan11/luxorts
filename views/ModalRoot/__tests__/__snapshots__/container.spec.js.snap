// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalRoot container renders ModalRoot component 1`] = `
<ModalRoot
  history={Object {}}
  modalProps={
    Object {
      "customProp": "customProp",
    }
  }
  modalType="MY_MODAL"
  onClose={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
